contest/templatetags/contest_extras.py

```diff
@@ -76,3 +77,22 @@ def users_sorted_by_solved_testcases(scoreboard):
 def total_contestant(scoreboard):
     return len(scoreboard.users)
 register.filter("total_contestant",total_contestant)
+
+@register.filter
+def can_register(user, contest_id):
+    try:
+        contest = Contest.objects.get(pk = contest_id)
+    except:
+        return False
+    return contest_info.can_register(user, contest)
+register.filter("can_register",can_register)
+
+@register.filter
+def show_register_btn(user, contest_id):
+    try:
+        contest = Contest.objects.get(pk = contest_id)
+    except:
+        return False
+    return (user_info.has_contest_ownership(user, contest) or 
+        contest_info.can_register(user, contest)) and not contest_info.is_ended(contest)
```

Comment:
I don't understand why the register button should show up for owner / coowners after a contest is end.